modules/live-schedule-impl/src/main/java/org/opencastproject/liveschedule/impl/LiveScheduleServiceImpl.java

```diff
@@ -303,77 +303,66 @@ public boolean updateLiveEventAcl(String mpId, AccessControlList acl) throws Liv
     // nothing to do. Note that this is just to save time; if the entry has already been deleted, the mp
     // will be compared below.
-    if (snapshot.getVersion().equals(snapshotVersionCache.getIfPresent(previousMp.getIdentifier().toString()))) {
+    if (snapshot.getVersion().equals(snapshotVersionCache.getIfPresent(mpId))) {
       logger.debug("Snapshot version {} was created by us so this change is ignored.", snapshot.getVersion());
       return false;
     }
-    // Temporary mp
-    MediaPackage tempMp = (MediaPackage) snapshot.getMediaPackage().clone();
-    // Set duration (used by live tracks)
-    setDuration(tempMp, episodeDC);
-    // Add live tracks to media package
-    Map<String, Track> generatedTracks = addLiveTracks(tempMp, episodeDC.getFirst(DublinCore.PROPERTY_SPATIAL));
-    // Update tracks in the publication
-    createOrUpdatePublicationTracks(tempMp, generatedTracks);
-    // If same mp, no need to do anything
-    if (isSameMediaPackage(previousMp, tempMp)) {
-      logger.debug("Live media package {} seems to be the same. Not updating.", previousMp);
+
+    // generate new live tracks
+    MediaPackage tmpMp = (MediaPackage) snapshot.getMediaPackage().clone();
+    setDurationForMediaPackage(tmpMp, episodeDC); // duration is used by live tracks
+    Map<String, Track> liveTracks = addLiveTracksToMediaPackage(tmpMp, episodeDC);
+
+    // if nothing changed, no need to do anything
+    if (isSameMediaPackage(mpFromSearch, tmpMp)) {
+      logger.debug("Live media package {} seems to be the same. Not updating.", mpFromSearch);
       return false;
     }
-    logger.info("Updating live media package {}", previousMp);
-    // Add and distribute catalogs/acl, this creates a new mp
-    MediaPackage mp = addAndDistributeElements(snapshot);
-    // Add tracks from tempMp
-    for (Track t : tempMp.getTracks()) {
-      mp.add(t);
-    }
-    // Remove publication element that came with the snapshot mp
-    removeLivePublicationChannel(mp);
-    // Publish mp to engage search index
-    publish(mp);
-    // Publication channel already there so no need to add
-    // Don't leave garbage there!
-    retractPreviousElements(previousMp, mp);
+
+    logger.info("Updating live media package {}", mpFromSearch);
+
+    // update mp in search
+    MediaPackage mpForSearch = distributeAclsAndCatalogs(snapshot);
+    for (Track t : tmpMp.getTracks()) {
+      mpForSearch.add(t);
+    }
+    removeLivePublicationChannel(mpForSearch); // we don't need the live publication in search
+    publishToSearch(mpForSearch);
+    retractPreviousElements(mpFromSearch, mpForSearch); // cleanup
+
+    // update live publication in archive
+    MediaPackage updatedArchivedMp = updateLivePublication(snapshot.getMediaPackage(), liveTracks);
+    snapshotVersionCache.put(mpId, assetManager.takeSnapshot(updatedArchivedMp).getVersion());
     return true;
   }
```

Comment:
@KatrinIhler So, this new line 364 and 365 is the crux of the patch to update the publication element in the archive when the live publication track changes. Previously, just the mediapackage top level metadata and catalog is updated but not the publication element, and this updates the publication element as well. Does that sound right?

Reply:
Yep!